api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -102,7 +102,8 @@ public boolean isUnpartitioned() {
     return !isPartitioned();
   }
 
-  int lastAssignedFieldId() {
+  // visible for testing
+  public int lastAssignedFieldId() {
```

Comment:
This is in the api module, so we might choose simply to not test this instead (or access the data by other means).

Reply:
I don't think that we need this change. This isn't something that we need to validate because it is derived from the fields for each `PartitionSpec`.

Reply:
Reverted this and am no longer testing it.